dali/python/nvidia/dali/plugin/jax/iterator.py

```diff
@@ -229,7 +229,12 @@ def _build_output_with_sharding(self, category_outputs):
         This output is compatible with automatic parallelization with JAX.
         """
         shard_shape = category_outputs[0].shape
-        global_shape = (self._num_gpus * shard_shape[0], *shard_shape[1:])
+
+        if isinstance(self._sharding, NamedSharding):
+            global_shape = (self._sharding.mesh.size * shard_shape[0], *shard_shape[1:])
+        else:
+            global_shape = (self._sharding.shape[0] * shard_shape[0], *shard_shape[1:])
```

Comment:
`sharding` variants have inconsisten APIs when it comes to getting the global shape.